core/unit_test/tools/TestWithoutInitializing.cpp

```diff
@@ -82,26 +107,46 @@ TEST(kokkosp, create_mirror_no_init_view_ctor) {
   Kokkos::View<int*, Kokkos::DefaultExecutionSpace> device_view("device view",
                                                                 10);
   Kokkos::View<int*, Kokkos::HostSpace> host_view("host view", 10);
+  using MemorySpace = typename Kokkos::DefaultExecutionSpace::memory_space;
+  MemorySpace device_memory_space;
 
   auto success = validate_absence(
       [&]() {
         auto mirror_device = Kokkos::create_mirror(
             Kokkos::view_alloc(Kokkos::HostSpace{},
                                Kokkos::WithoutInitializing),
             device_view);
+        static_assert(
+            std::is_same_v<typename decltype(mirror_device)::memory_space,
+                           Kokkos::HostSpace>);
         auto mirror_host = Kokkos::create_mirror(
-            Kokkos::view_alloc(Kokkos::HostSpace{}, Kokkos::WithoutInitializing,
+            Kokkos::view_alloc(device_memory_space, Kokkos::WithoutInitializing,
                                Kokkos::DefaultExecutionSpace{}),
             host_view);
+    // FIXME_MSVC
```

Comment:
why would this fail on MSVC?

Reply:
ad a minimum we need a comment whats going on.

Reply:
I missed removing those as well in the last comment. The issue was `decltype` returning a reference for `MSVC.